src/components/sets/SetsPage.jsx

```diff
@@ -1,17 +1,173 @@
+import { ResponsivePie } from "@nivo/pie";
+import { ResponsiveBar } from "@nivo/bar";
+import {
+  getColorData,
+  getCardPriceData,
+  getRarityData,
+  getCardTypesData,
+} from "./SetsFunctions.jsx";
 
-export const SetsPage = () => {
+export const SetsPage = ({ setInfo, loadingSets: isLoading, errorSets: loadError }) => {
+  const [selectedSet, setSelectedSet] = useState("");
+  const [iconUri, setIconUri] = useState("");
+  const navigate = useNavigate();
+  if (loadError) {
+    navigate("/error", { state: { errorMessages: loadError.message } });
+  }
+
+  useEffect(() => {
+    if (setInfo.length > 0) {
+      const firstSet = setInfo[0];
+      setSelectedSet(firstSet.code); // Set the default selected set code
+      setIconUri(firstSet.icon_svg_uri); // Set the default icon URI
+    }
+  }, [setInfo]);
+
+  const handleSelectChange = (e) => {
+    const selectedCode = e.target.value;
+    setSelectedSet(selectedCode); // Update the selected set state
+
+    // Find the corresponding set from setInfo and update the iconUri
+    const selectedSetInfo = setInfo.find((set) => set.code === selectedCode);
+    if (selectedSetInfo) {
+      setIconUri(selectedSetInfo.icon_svg_uri);
+    }
+  };
+
+  //Load data from the API
+  const { loading, data } = useQuery(GET_SETS_BY_CODE(selectedSet));
+  if (loading) {
+    return "Loading...";
+  }
+
+  //Fetch data from functions
+  const colorData = getColorData(data);
+  const priceData = getCardPriceData(data);
+  const rarityData = getRarityData(data);
+  const typesData = getCardTypesData(data);
+
+  //Page Setup
   return (
     <>
       <MainNavigation />
       <Container fluid>
         <Row>
           <Col>
-            <h1 className={"mt-2"}>Sets</h1>
+            <h1 className={"mt-2"}>Set Data</h1>
           </Col>
         </Row>
         <hr />
+        {isLoading && (
+          <div className="d-flex justify-content-center align-items-center">
+            <Spinner animation="border" role="status">
+              <span className="visually-hidden">Loading Set Data...</span>
+            </Spinner>
+          </div>
+        )}
+        {!isLoading && (
+          <Row>
+            <Col md={2}>
+              <Form.Select value={selectedSet} onChange={handleSelectChange}>
+                {setInfo.map(({ code, name }) => (
+                  <option key={code} value={code}>
+                    {name}
+                  </option>
+                ))}
+              </Form.Select>
+            </Col>
+          </Row>
+        )}
+        {selectedSet && (
+          <Row>
+            <Col>
+              {iconUri && (
+                <img
+                  src={iconUri}
+                  alt="Selected Set Icon"
+                  style={{ width: "100px", height: "100px" }}
+                />
+              )}
+              <p>Selected Set Code: {selectedSet}</p>
+            </Col>
+          </Row>
+        )}
+        {selectedSet && (
+          <Row>
+            <div>
```

Comment:
Wrapping each graph div in a `<Col lg={6} className={"m-0'}> looked pretty good to me. 2 graphs on larger screens and 1 per row on smaller screens. The margins could probably be messed with